release notes/v0.29.0.md

```diff
@@ -0,0 +1,133 @@
+echo '{"dns": {"select": "roundRobin", "ttl": "1h33m7s", "policy": "onlyIPv6"}}' > config.json
+k6 run --config "config.json" script.js
+```
+
+### Support for Go extensions ([#1688](https://github.com/loadimpact/k6/pull/1688))
+
+After some discussions ([#1353](https://github.com/loadimpact/k6/issues/1353)) and exploration of different approaches for Go-based k6 extensions, we've settled on adopting an approach similar to [caddy's](https://caddyserver.com/docs/extending-caddy). In short, [`xk6`](https://github.com/k6io/xk6) (modeled after [`xcaddy`](https://github.com/caddyserver/xcaddy)) is a small stand-alone tool that will allow the building of custom k6 binaries with 3rd party extensions bundled in. The extensions can be simple Git repositories (no central infrastructure needed!) with Go modules. They will be fully compiled, not interpreted, a part of the final custom k6 binary users will be able to build with k6.
+
+xk6 is not yet stable or documented, so any plugin authors will struggle until we stabilize and document everything in the coming weeks. The important part is that the k6 changes that would allow xk6 to work were implemented in [#1688](https://github.com/loadimpact/k6/pull/1688), so k6 v0.29.0 is the first version compatible with xk6!
+
+Expect more information soon, but for a brief example, xk6 will work somewhat like this:
+
+```bash
+xk6 build v0.29.0 --with github.com/k6io/xk6-k8s --with github.com/k6io/xk6-sql@v0.1.1
+
+./k6 run some-script-with-sql-and-k8s.js
+```
+
+Thanks, @andremedeiros, for pushing us to add plugins in k6 and for making a valiant attempt to harness Go's poor plugin API! Thank you, @mardukbp, for pointing us towards xcaddy and explaining its benefits!
+
+### Support for setting local IPs, potentially from multiple NICs ([#1682](https://github.com/loadimpact/k6/pull/1682))
+
+You can now specify a list of source IPs, IP ranges and CIDRs for `k6 run`, from which VUs will make requests via the new `--local-ips` CLI flag or `K6_LOCAL_IPS` environment variable. The IPs will be sequentially given out to VUs, allowing you to distribute load between different local addresses. This option doesn't change anything on the OS level, so the IPs need to already be configured on the OS level in order for k6 to be able to use them.
+
+The biggest use case for this feature is splitting the network traffic from k6 between multiple network adapters, thus potentially greatly increasing the available network throughput. For example, if you have 2 NICs, you can run k6 with `--local-ips="<IP-from-first-NIC-1>,<IP-from-first-NIC-2>"` to balance the traffic equally between them - half of the VUs will use the first IP and the other half will use the second. This can scale to any number of NICs, and you can repeat some local IPs to give them more traffic. For example, `--local-ips="<IP1>,<IP2>,<IP3>,<IP3>"` will split VUs between 3 different source IPs in a 25%:25%:50% ratio.
```

Comment:
```suggestion
The biggest use case for this feature is splitting the network traffic from k6 between multiple network adapters, thus potentially greatly increasing the available network throughput. For example, if you have 2 NICs, you can run k6 with `--local-ips="<IP-from-first-NIC>,<IP-from-second-NIC>"` to balance the traffic equally between them - half of the VUs will use the first IP and the other half will use the second. This can scale to any number of NICs, and you can repeat some local IPs to give them more traffic. For example, `--local-ips="<IP1>,<IP2>,<IP3>,<IP3>"` will split VUs between 3 different source IPs in a 25%:25%:50% ratio.
```

?